test/FlowChartSpec.hs

```diff
@@ -0,0 +1,84 @@
+                id
+            |]
+      parseMaybe pDiagram diagram
+        `shouldHaveEdges` []
+    it "nodes of single edge with two nodes" $ do
+      let diagram =
+            [text|
+              flowchart TD
+                start --> stop
+            |]
+      parseMaybe pDiagram diagram
+        `shouldHaveNodes` [ Node {nodeId = NodeId {unNodeId = "start"}, nodeShape = Nothing, nodeLabel = Nothing},
+                            Node {nodeId = NodeId {unNodeId = "stop"}, nodeShape = Nothing, nodeLabel = Nothing}
+                          ]
+    it "edges of single edge with two nodes" $ do
+      let diagram =
+            [text|
+              flowchart TD
+                start --> stop
+            |]
+      parseMaybe pDiagram diagram
+        `shouldHaveEdges` [(Just (Edge {edgeStyle = Just "-->", edgeLabel = Nothing}), Node {nodeId = NodeId {unNodeId = "start"}, nodeShape = Nothing, nodeLabel = Nothing}, Node {nodeId = NodeId {unNodeId = "stop"}, nodeShape = Nothing, nodeLabel = Nothing})]
+    it "nodes of single edge with two times two isolated nodes" $ do
+      let diagram =
+            [text|
+              flowchart TD
+                A & B--> C & D
+            |]
+      parseMaybe pDiagram diagram
+        `shouldHaveNodes` [ Node {nodeId = NodeId {unNodeId = "A"}, nodeShape = Nothing, nodeLabel = Nothing},
+                            Node {nodeId = NodeId {unNodeId = "B"}, nodeShape = Nothing, nodeLabel = Nothing},
+                            Node {nodeId = NodeId {unNodeId = "C"}, nodeShape = Nothing, nodeLabel = Nothing},
+                            Node {nodeId = NodeId {unNodeId = "D"}, nodeShape = Nothing, nodeLabel = Nothing}
+                          ]
+    it "edges of single edge with two times two isolated nodes" $ do
+      let diagram =
+            [text|
+              flowchart TD
+                A & B--> C & D
+            |]
+      parseMaybe pDiagram diagram
+        `shouldHaveEdges` [ (Just (Edge {edgeStyle = Just "-->", edgeLabel = Nothing}), Node {nodeId = NodeId {unNodeId = "A"}, nodeShape = Nothing, nodeLabel = Nothing}, Node {nodeId = NodeId {unNodeId = "C"}, nodeShape = Nothing, nodeLabel = Nothing}),
+                            (Just (Edge {edgeStyle = Just "-->", edgeLabel = Nothing}), Node {nodeId = NodeId {unNodeId = "A"}, nodeShape = Nothing, nodeLabel = Nothing}, Node {nodeId = NodeId {unNodeId = "D"}, nodeShape = Nothing, nodeLabel = Nothing}),
+                            (Just (Edge {edgeStyle = Just "-->", edgeLabel = Nothing}), Node {nodeId = NodeId {unNodeId = "B"}, nodeShape = Nothing, nodeLabel = Nothing}, Node {nodeId = NodeId {unNodeId = "C"}, nodeShape = Nothing, nodeLabel = Nothing}),
+                            (Just (Edge {edgeStyle = Just "-->", edgeLabel = Nothing}), Node {nodeId = NodeId {unNodeId = "B"}, nodeShape = Nothing, nodeLabel = Nothing}, Node {nodeId = NodeId {unNodeId = "D"}, nodeShape = Nothing, nodeLabel = Nothing})
+                          ]
```

Comment:
Again, with the lens/IsString tip above, this would start to look like:

```haskell
parseMaybe pDiagram diagram
  `shouldHaveEdges` [ (fromString "-->", fromString "A", fromString "C"),
  ...
````

Reply:
Should we keep the `Just` constructor for `-->` ? Because i think `(Nothing, "A", "C")` is a possible scenario.